Add OkPacket interface for inserts, updates, and deletes

sedenardi opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mysqljs/mysql/blob/master/lib/protocol/packets/OkPacket.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

This PR adds the interface for the `OkPacket` object type that's returned on insert, update, and delete statements. As you can see from the [mysql source](https://github.com/mysqljs/mysql/blob/master/lib/protocol/packets/OkPacket.js), I did not include all fields. I only included fields which were documented in the library's README, and I used those fields' documentation in the type definition.

kpping commented:
Thank you for your PR @sedenardi.

I think if we want to change from packetCallback to OkPacket we must include all fields because it might break someone project that relied on those neglect fields. 

Could you please re-submit the PR with all fields ?

Thank you.

sedenardi commented:
Thanks for taking a look @kpping.

Looking through the fields on [`Query`](https://github.com/mysqljs/mysql/blob/master/lib/protocol/sequences/Query.js#L62) I see that the type of the `OkPacket` member isn't actually the `OkPacket` type but the `packetCallback` function (which takes in an `OkPacket` as its argument), so I've reverted that change.

I've also added in the rest of the fields on the `OkPacket` type itself. `serverStatus` and `warningCount` are [conditionally](https://github.com/mysqljs/mysql/blob/master/lib/protocol/packets/OkPacket.js#L22) set so I made them optional.
